### PR TITLE
Convert all plugins to the use of the new Plugins::InterfaceBase class.

### PR DESCRIPTION
**Pull request by bangerth**
This is the follow-up to #1807 that converts all plugins I could find to derive from the abstract base class. It allows removing almost all function declarations of boiler-plate functions in the plugin declarations because they are now provided by the base class.

I will rebase once #1807 is merged.

### CODE REVIEW

File: include/aspect/particle/generator/interface.h [hunk 1]
```diff
@@ -65,43 +65,27 @@ namespace aspect
...
-           * should generate the particles and associate them to their according
-           * cells by inserting them into a multimap between cell and particle.
-           * This map becomes very large if the particle count per process
-           * is large, so we hand it over by reference instead of returning
-           * the multimap.
-           *
-           * @param [in,out] particles A multimap between cells and their
-           * particles. This map will be filled in this function.
-           */
+             * Generate particles. Every derived class
```
Comment (tjhei) [inline include/aspect/particle/generator/interface.h:76]: why this change?

**Review by tjhei — APPROVED** [review]
Those were a *lot* of duplicated lines.

**Comment by bangerth** [comment]
I also pushed a commit that adds the missing documentation flagged in #1807.

**Review by gassmoeller — APPROVED** [review]
I like it!
